src/lug/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1,12 +1,203 @@
+    if (!_fullscreen)
+    {
+        RECT rectangle = { 0, 0, _parent->_mode.width, _parent->_mode.height };
+        AdjustWindowRect(&rectangle, win32Style, false);
+        realWidth = rectangle.right - rectangle.left;
+        realHeight = rectangle.bottom - rectangle.top;
+    }
+
+    // Create the window
+    {
+        std::vector<wchar_t> realTitle(title.length() + 1); // Convert a const char * to a wchar_t
+        mbstowcs(&realTitle[0], title.c_str(), title.length() + 1);
+
+        _handle = CreateWindowW(_className, &realTitle[0], win32Style, left, top, realWidth, realHeight, NULL, NULL, GetModuleHandle(NULL), this);
+    }
+
+    // Switch to fullscreen if requested
+    if (_fullscreen)
+    {
+        setFullscreen();
+    }
+
+    // Increment window count
+    _windowCount++;
+    return true;
+}
+
+void lug::Window::priv::WindowImpl::close() {
+    cleanup();
+}
+
+bool lug::Window::priv::WindowImpl::pollEvent(lug::Window::Event& event) {
+    // We process the window events only if we own it
+    if (!_callback)
+    {
+        MSG message;
+        while (PeekMessageW(&message, NULL, 0, 0, PM_REMOVE))
+        {
+            TranslateMessage(&message);
+            DispatchMessageW(&message);
+        }
+    }
+    if (!_events.empty()) {
+        event = _events.front();
+        _events.pop();
+        return true;
+    }
+    return false;
+}
+
+void lug::Window::priv::WindowImpl::processWindowEvents(UINT message, WPARAM wParam, LPARAM lParam) {
+    lug::Window::Event e;
+
+    switch (message) {
+    case WM_CLOSE:
+        e.type = EventType::CLOSE;
+        break;
+    case WM_DESTROY:
+        e.type = EventType::DESTROY;
+        break;
+    default:
+        return;
+    }
+    _events.push(std::move(e));
+}
+
+void lug::Window::priv::WindowImpl::registerWindow() {
+    // Register the window class with windows
+    WNDCLASSW windowClass =
+    {
+        windowClass.style = 0,
+        windowClass.lpfnWndProc = &lug::Window::priv::WindowImpl::onEvent,
+        windowClass.cbClsExtra = 0,
+        windowClass.cbWndExtra = 0,
+        windowClass.hInstance = GetModuleHandleW(NULL),
+        windowClass.hIcon = NULL,
+        windowClass.hCursor = 0,
+        windowClass.hbrBackground = 0,
+        windowClass.lpszMenuName = NULL,
+        windowClass.lpszClassName = _className
+    };
+    RegisterClassW(&windowClass);
+}
+
+void lug::Window::priv::WindowImpl::setFullscreen() {
+    DEVMODEW devMode;
+    devMode.dmSize = sizeof(devMode);
+    devMode.dmPelsWidth = _parent->_mode.width;
+    devMode.dmPelsHeight = _parent->_mode.height;
+    devMode.dmBitsPerPel = 32;
+    devMode.dmFields = DM_PELSWIDTH | DM_PELSHEIGHT | DM_BITSPERPEL;
+
+    // Apply fullscreen mode
+    if (ChangeDisplaySettingsW(&devMode, CDS_FULLSCREEN) != DISP_CHANGE_SUCCESSFUL)
+    {
+        std::cerr << "Failed to change display mode for fullscreen" << std::endl;
```

Comment:
Don't use std::cerr! Return `false` and handle it in the `create` function?
I think that's okay to not fail the creation of the window and just tell it to the user (with the logger latter)